src/lib/components/ToastAlert/useToastManager.ts

```diff
@@ -0,0 +1,111 @@
+import { useCallback, useContext, useMemo } from "react"
+import { AlertContext } from "./ToastAlertProvider"
+import { TAlert, TNewToastOptions } from "./ToastAlert.model"
+import { originCalculation } from "./ToastAlert.utils"
+
+export function useToastManager() {
+  const { dispatch, toastAlerts, origin } = useContext(AlertContext)
+
+  const defaultDuration = useMemo(() => 7000, [])
+  const defaultIntent = useMemo(() => "primary" as const, [])
+  const originTranslate = useMemo(() => originCalculation(origin), [origin])
+
+  const removeToastTimeout = useCallback(
+    (id: TAlert["id"] | undefined) => {
+      const alert = toastAlerts.find(alert => alert.id === id)
+      clearTimeout(alert?.timeout)
+    },
+    [toastAlerts],
+  )
+
+  const removeAlert = useCallback(
+    (id: TAlert["id"]) => dispatch({ type: "REMOVE_ALERT", payload: { id } }),
+    [dispatch],
+  )
+
+  const animateToast = useCallback(
+    async (alert: Element, onFinishCb: () => void) => {
+      const slideFrames = new KeyframeEffect(alert, [{ translate: originTranslate }], {
+        duration: 300,
+        easing: "ease",
+        fill: "forwards",
+      })
+
+      const { height } = alert.getBoundingClientRect()
+      const collapseFrames = new KeyframeEffect(
+        alert,
+        [{ maxHeight: `${height}px` }, { maxHeight: "0" }],
+        { duration: 150, easing: "ease-in", fill: "forwards" },
+      )
+      const slideAnimation = new Animation(slideFrames, document.timeline)
+      const collapseAnimation = new Animation(collapseFrames, document.timeline)
+      slideAnimation.play()
+      await slideAnimation.finished
+      collapseAnimation.play()
+      await collapseAnimation.finished
+      onFinishCb()
+    },
```

Comment:
Remove this callback dependency and call it on `.then` at the call site of `animateToast`